source/val/BasicBlock.h

```diff
@@ -61,22 +75,40 @@ class BasicBlock {
   /// Returns the successors of the BasicBlock
   std::vector<BasicBlock*>* get_successors() { return &successors_; }
 
-  /// Returns true if the  block should be reachable in the CFG
+  /// Returns true if the block is reachable in the CFG
   bool is_reachable() const { return reachable_; }
 
+  /// Returns the type of the BasicBlock
+  bool is_type(BlockType type) const { return type_.test(type); }
+
+  /// Sets the reacability of the basic block in the CFG
```

Comment:
Minor typo in comment.  "reachability".
Minor:  I could also go with "set_reachable"  since it matches the member name.